Modules/_datetimemodule.c

```diff
@@ -3234,6 +3254,9 @@ date_isocalendar(PyDateTime_Date *self, PyObject *Py_UNUSED(ignored))
     int  week;
     int  day;
 
+    PyObject* v = PyStructSequence_New(&StructIsoCalendarDateType);
```

Comment:
```suggestion
    PyObject *v = PyStructSequence_New(&StructIsoCalendarDateType);
```
Also, I think this `PyStructSequence_New()` call should be moved down (so that it is closer to the `PyStructSequence_SET_ITEM()` calls) .